app/controllers/api/instances_controller.rb

```diff
@@ -140,8 +140,12 @@ def pause_instance(instance)
     end
 
     def validate_instance_for_action(instance, action)
-      validation = instance.send("validate_#{action}")
-      action_result(validation[:available], validation[:message].to_s)
+      if instance.respond_to?("supports_#{action}?")
+        action_result(instance.public_send("supports_#{action}?"), instance.unsupported_reason(action.to_sym))
```

Comment:
Now that https://github.com/ManageIQ/manageiq/pull/10917 is merged, you can get rid of one `public_send`

```ruby
action_result(instance.supports?(action), instance.unsupported_reason(action.to_sym))
```

I still have to come up with a query method for checking if a feature is generally queriable.